### PR TITLE
Debug mode

### PR DESCRIPTION
**Pull request by silviu-toderita**
@adityab82 When you have a chance, can you try flashing this branch to your steering wheel? I want to see if it makes the behaviour of the proximity sensor more consistent. It contains some code to print the value coming from the proximity sensor, just uncomment the *#define DEBUG_MODE* line near the top of main.cpp. That will also output the value of the throttle pedal, which will be meaningless until we connect it to the car. 

I'm very curious to know if the behaviour of the proximity sensor is the same between different steering wheels, or if each one needs its own calibration value. It can be calibrated by changing *DMS_DELTA* and *DMS_DELTA_COUNT*. DMS Delta adjusts how much higher the signal from the proximity sensor should be when the LED is on vs when the LED is off. DMS Delta Count is the number of cycles that the delta value has to be above the threshold before the "DMS" light turns on, to prevent noise.

### CODE REVIEW

**Comment by silviu-toderita** [comment]
@adityab82 @cosparks I've finally fixed the DMS and throttle. Here's what the issue was:
- The DMS LED was never turning on. That's why the difference between LED on and LED off was so tiny. Now that the LED turns on, a threshold of 200 seems to work well. 
- There was some funky math and incorrect types happening with the throttle output. It's been fixed.

**Comment by silviu-toderita** [comment]
Also @adityab82 can you try this on your steering wheel before we merge it? I want to make sure 200 is a good value for the `DMS_DELTA_THRESHOLD` for both steering wheels.

**Comment by cosparks** [comment]
Curious now what the issue with the math was.  What caused it to fail?

**Comment by silviu-toderita** [comment]
@cosparks there was both an overflow and an incorrect variable name (that second one was probably introduced on test day when frantically debugging) where we were returning the raw throttle input (0-5000ish) as a uint8_t

**Comment by cosparks** [comment]
@silviu-toderita ohh so the output wasn't being scaled before being cast to uint8_t (throttle_t)?  Sorry if it wasn't obvious with the way I set up those types.  I was thinking that it would be good to be able to just change those typedefs if for instance we wanted higher resolution for throttle output (uint16_t as opposed to uint8_t), but there's definitely a tradeoff with readability.

File: src/main.cpp [hunk 1]
```diff
@@ -202,25 +206,41 @@ void handle_motor_inputs(){
...
 	}
 
-	return (throttle_t)(throttleAsFloat * SLOPE + OFFSET);
+	// Normalize throttle value
+	throttleVal -= MIN_THROTTLE_INPUT;
+
+	// Scale for output
+	throttle_t throttleOut = (throttleVal * MAX_THROTTLE_OUTPUT) / THROTTLE_RANGE;
+
+	return throttleOut;
```
Comment (cosparks) [inline src/main.cpp:229]: Cool, just wrote a little test in an online gcc compiler and this works great (unsurprisingly)

https://onlinegdb.com/HN19vJ9cA

File: src/main.cpp [hunk 2]
```diff
@@ -202,25 +206,41 @@ void handle_motor_inputs(){
...
 }
 
 data_t getDmsVal() {
-	float dms_ctrl = dms.read();
+	int dmsCtrl = (int)(dms.read() * 10000);
 	dmsLed.write(1);
 	wait_us(40);
-    float dms_val = dms.read();
+    int dmsVal = (int)(dms.read() * 10000);
```
Comment (cosparks) [inline src/main.cpp:236]: Makes sense.  What was the reason to switch to using int on [0,10000] as opposed to floats on [0,1]?

**Review by cosparks — APPROVED** [review]

File: src/main.cpp (hunk 2 above)
Comment (silviu-toderita) [reply src/main.cpp:236]: i hate floats

Comment (cosparks) [reply src/main.cpp:236]: ok

**Comment by adityab82** [comment]
@silviu-toderita it is working well on my steering as well. It starts detecting my hand from about 3-4 cm off the steering

Update: I am getting some flickering (false positives) when kept about 15-20 cm away from a wall.

The value 250 is working well for my steering

**Pull request merged**